lib/matplotlib/tests/test_font_manager.py

```diff
@@ -253,11 +253,16 @@ def _test_threading():
     from matplotlib.ft2font import LOAD_NO_HINTING
     import matplotlib.font_manager as fm
 
+    def loud_execepthook(arg):
+        raise RuntimeError("error in thread!")
+
+    threading.excepthook = loud_execepthook
+
     N = 10
     b = threading.Barrier(N)
 
     def bad_idea(n):
-        b.wait()
+        b.wait(timeout=.5)
```

Comment:
Since the total timeout is 10s, this could probably be a bit longer, maybe 5s, just for slower systems?